usaspending_api/broker/management/commands/fabs_nightly_loader.py

```diff
@@ -0,0 +1,305 @@
+
+# start = timeit.default_timer()
+# function_call
+# end = timeit.default_timer()
+# time elapsed = str(end - start)
+
+
+logger = logging.getLogger('console')
+exception_logger = logging.getLogger("exceptions")
+
+subtier_agency_map = {
+    subtier_agency['subtier_code']: subtier_agency['subtier_agency_id']
+    for subtier_agency in SubtierAgency.objects.values('subtier_code', 'subtier_agency_id')
+    }
+subtier_to_agency_map = {
+    agency['subtier_agency_id']: {'agency_id': agency['id'], 'toptier_agency_id': agency['toptier_agency_id']}
+    for agency in Agency.objects.values('id', 'toptier_agency_id', 'subtier_agency_id')
+    }
+toptier_agency_map = {
+    toptier_agency['toptier_agency_id']: toptier_agency['cgac_code']
+    for toptier_agency in ToptierAgency.objects.values('toptier_agency_id', 'cgac_code')
+    }
+
+award_update_id_list = []
+
+
+class Command(BaseCommand):
+    help = "Update FABS data nightly"
+
+    @staticmethod
+    def get_fabs_data(db_cursor, date):
+        # The ORDER BY is important here because deletions must happen in a specific order and that order is defined
+        # by the Broker's PK since every modification is a new row
+        db_query = 'SELECT * ' \
+                   'FROM published_award_financial_assistance ' \
+                   'WHERE created_at >= %s' \
+                   'ORDER BY published_award_financial_assistance_id ASC'
+        db_args = [date]
+
+        db_cursor.execute(db_query, db_args)
+        db_rows = dictfetchall(db_cursor)  # this returns an OrderedDict
+
+        ids_to_delete = []
+
+        # Iterate through the result dict and determine what needs to be deleted and what needs to be added
+        for row in db_rows:
+            if row['correction_late_delete_ind'] and row['correction_late_delete_ind'].upper() == 'D':
+                ids_to_delete += [row['published_award_financial_assistance_id']]
+                # remove the row from the list of rows from the Broker since once we delete it, we don't care about it.
+                # all that'll be left in db_rows are rows we want to insert
+                db_rows.remove(row)
+
+        logger.info('Number of records to insert/update: %s' % str(len(db_rows)))
+        logger.info('Number of records to delete: %s' % str(len(ids_to_delete)))
+
+        return db_rows, ids_to_delete
+
+    @staticmethod
+    def delete_stale_fabs(ids_to_delete=None):
+        logger.info('Starting deletion of stale FABS data')
+
+        if not ids_to_delete:
+            return
+
+        # This cascades deletes for TransactionFABS & Awards in addition to deleting TransactionNormalized records
+        TransactionNormalized.objects.\
+            filter(assistance_data__published_award_financial_assistance_id__in=ids_to_delete).delete()
+
+    @staticmethod
+    def insert_new_fabs(to_insert, total_rows):
+        logger.info('Starting insertion of new FABS data')
+
+        place_of_performance_field_map = {
```

Comment:
Broker FABS now has pop_county_code and pop_country name:
column names are `place_of_perform_country_n` and `place_of_perform_country_c`

Reply:
Will add